backend/de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/ddl/functions/update_FieldTranslation_From_AD_Name_Element.sql

```diff
@@ -1,91 +1,100 @@
-		Name = x.Name,
-		Description = x.Description,
-		Help = x.Help
-	FROM
-	(
-		select
-			f.AD_Field_ID,
-			etrl.AD_Element_ID,
-			etrl.AD_Language,
-			etrl.IsTranslated,
-			etrl.Name,
-			etrl.Description,
-			etrl.Help
-		from AD_Element_Trl_Effective_v etrl
-			join AD_Field f on f.AD_Name_ID = etrl.AD_Element_ID
-		where 
-			etrl.AD_Element_ID = p_AD_Element_ID
-			and (p_AD_Language is null OR etrl.AD_Language = p_AD_Language)
-	) x
-	WHERE
-		t.AD_Field_ID = x.AD_Field_ID
-		and t.AD_Language = x.AD_Language
-	;
-	--
-	GET DIAGNOSTICS update_count_via_AD_Name = ROW_COUNT;
-
-	raise notice 'Updated AD_Field_Trl for AD_Element_ID=%, AD_Language=%: % rows via AD_Column, % rows via AD_Name',
-		p_AD_Element_ID, p_AD_Language, update_count_via_AD_Column, update_count_via_AD_Name;
-	
+    --
+    -- AD_Field_Trl via AD_Column
+    UPDATE AD_Field_Trl f_trl
+    SET IsTranslated = e_trl.IsTranslated,
+        Name         = e_trl.Name,
+        Description  = e_trl.Description,
+        Help         = e_trl.Help
+    FROM AD_Element_Trl_Effective_v e_trl
+          WHERE( p_AD_Element_ID IS NULL OR e_trl.AD_Element_ID = p_AD_Element_ID)
+            AND (p_AD_Language IS NULL OR e_trl.AD_Language = p_AD_Language)
+            AND f_trl.ad_language = e_trl.ad_language
+
+            AND EXISTS(SELECT 1 FROM ad_column c
+                                JOIN AD_Field f on c.ad_column_id = f.ad_column_id
+                                WHERE c.ad_element_id = e_trl.ad_element_id
+                                  AND f_trl.ad_field_id = f.ad_field_id
+                AND f.ad_name_id IS NULL);
+    --
+    GET DIAGNOSTICS update_count_via_AD_Column = ROW_COUNT;
+
+
+    --
+    -- AD_Field_Trl via AD_Element -> AD_Name_ID
+    UPDATE AD_Field_Trl f_trl
+    SET IsTranslated = e_trl.IsTranslated,
+        Name         = e_trl.Name,
+        Description  = e_trl.Description,
+        Help         = e_trl.Help
+
+          FROM AD_Element_Trl_Effective_v e_trl
+    WHERE( p_AD_Element_ID IS NULL OR e_trl.AD_Element_ID = p_AD_Element_ID)
+      AND (p_AD_Language IS NULL OR e_trl.AD_Language = p_AD_Language)
+      AND f_trl.ad_language = e_trl.ad_language
+      AND EXISTS(SELECT 1 FROM AD_Field f
+                          WHERE f.ad_name_id = e_trl.ad_element_id
+          AND f.ad_field_id = f_trl.ad_field_id);
+
+    --
+    GET DIAGNOSTICS update_count_via_AD_Name = ROW_COUNT;
+
+    RAISE NOTICE 'Updated AD_Field_Trl for AD_Element_ID=%, AD_Language=%: % rows via AD_Column, % rows via AD_Name',
+        p_AD_Element_ID, p_AD_Language, update_count_via_AD_Column, update_count_via_AD_Name;
+
+
+    IF (p_AD_Language IS NULL OR isBaseAD_Language(p_AD_Language) = 'Y') THEN
+
+
+        UPDATE AD_Field f
+        SET Name        = e_trl.Name,
+            Description = e_trl.Description
+        FROM AD_Element_Trl_Effective_v e_trl
+        WHERE (p_AD_Element_ID IS NULL OR e_trl.AD_Element_ID = p_AD_Element_ID)
```

Comment:
i think we also need `AND f.ad_name_id IS NULL`

Reply:
yes! Thanks!